File: JAVA/ProductoVectores.java
package JAVA;
import java.util.Arrays;
import java.util.Scanner;
class ProductoVector{
    public static void main(String[] args) {
        
        Scanner entrada = new Scanner(System.in); 
        
        System.out.println("El tamaño de los vectores a y b : ");
        int cantidad = entrada.nextInt();
        int[] a=new int[cantidad];   
        for (int i=0; i<=cantidad-1;i++){
            System.out.println();
            a[i]=entrada.nextInt();
        }
        int[] b=new int[cantidad];   
        for (int i=0; i<=cantidad-1;i++){
            b[i]=entrada.nextInt();
        }
        int[] producto=new int[cantidad];   
        for (int i=0; i<=cantidad-1;i++){
            producto[i]=a[i]*b[i];
        }

        System.out.println( Arrays.stream(producto).sum());


        
    }

}

File: Python/ProductoVectores.py
a = []
b = []
cantidad = int(input(f"Ingresa el tamaño de los vectores: "))
for i in range(cantidad):
    # Recuerda que range comenzará desde 0, así que imprimimos el número solicitado pero + 1
    numero = input(f"Ingresa el número {i + 1} del vector a: ")
    # Convertir a entero, pues input regresa una cadena
    numero = int(numero)
    # Lo agregamos al arreglo con append
    a.append(numero) 

for i in range(cantidad):
    # Recuerda que range comenzará desde 0, así que imprimimos el número solicitado pero + 1
    numero = input(f"Ingresa el número {i + 1} del vector b: ")
    # Convertir a entero, pues input regresa una cadena
    numero = int(numero)
    # Lo agregamos al arreglo con append
    b.append(numero) 

print(a,b)

productoPunto=[0]*cantidad
for i in range(cantidad):
    productoPunto[i]=a[i]*b[i]

print(sum(productoPunto))